tests/drivers/build_all/sensor/src/generic_test.c

```diff
@@ -176,36 +177,71 @@ static void run_generic_test(const struct device *dev)
 		/* Release the CQE */
 		rtio_cqe_release(&sensor_read_rtio_ctx, cqe);
 
-		enum sensor_channel channel;
-		sensor_frame_iterator_t fit = {0};
-		sensor_channel_iterator_t cit = {0};
 		const struct sensor_decoder_api *decoder;
-		int8_t shift;
-		q31_t q;
+		union sensor_data_union decoded_data;
```

Comment:
Why not just use a buffer here? Seems like the union basically just makes sure that the size is good, you can probably just allocate 512 Bytes and check it against the size info before you decode. This will also look a lot more like how the API is supposed to be used.

Reply:
This is gets the size just right and is cleaner IMO, esp since the comparison code below needs to handle multiple different representations based on channel type.